src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.UnitConverter/Main.cs

```diff
@@ -56,8 +57,51 @@ public List<Result> Query(Query query)
                 .ToList();
         }
 
+        private string GetOunceUnitNameString(UnitConversionResult result)
+        {
+            return result.UnitNameFrom switch
+            {
+                "usounce" => "US ounce",
+                "imperialounce" => "Imperial ounce",
+                _ => string.Empty,
+            };
+        }
+
         private Result GetResult(UnitConversionResult result)
         {
+            if (result.UnitNameFrom == "usounce" || result.UnitNameFrom == "imperialounce")
+            {
+                string ounceName = GetOunceUnitNameString(result);
+                return new Result
+                {
+                    ContextData = result,
+                    Title = result.ToString(null),
+                    IcoPath = _icon_path,
+                    Score = 300,
+                    SubTitle = string.Format(CultureInfo.CurrentCulture, ConvertFromOunce, ounceName) + " - " + string.Format(CultureInfo.CurrentCulture, CopyToClipboard, result.QuantityInfo.Name),
+                    Action = c =>
+                    {
+                        var ret = false;
+                        var thread = new Thread(() =>
+                        {
+                            try
+                            {
+                                Clipboard.SetText(result.ConvertedValue.ToString(UnitConversionResult.Format, CultureInfo.CurrentCulture));
+                                ret = true;
+                            }
+                            catch (ExternalException)
+                            {
+                                MessageBox.Show(Properties.Resources.copy_failed);
+                            }
+                        });
+                        thread.SetApartmentState(ApartmentState.STA);
+                        thread.Start();
+                        thread.Join();
+                        return ret;
+                    },
+                };
+            }
```

Comment:
Too have less duplicated code:

Then we need only one `Result` definition where we use the subTitleText variable.

```suggestion
            string unitFromName = GetFromUnitNameString(result);
            if (!string.IsNullOrEmpty(unitFromName))
            {
                    subTitleText = string.Format(CultureInfo.CurrentCulture, ConvertFromOunce, ounceName) + " - " + string.Format(CultureInfo.CurrentCulture, CopyToClipboard, result.QuantityInfo.Name);
            }
            else
            {
                    subTitleText = string.Format(CultureInfo.CurrentCulture, CopyToClipboard, result.QuantityInfo.Name);
            }
```